ql/src/java/org/apache/hadoop/hive/ql/DriverState.java

```diff
@@ -89,7 +89,11 @@ public void compiling() {
   public void compilingWithLocking() {
     lock();
     try {
-      driverState = State.COMPILING;
+      if (isDestroyed() || isClosed()) {
+        abort();
```

Comment:
`abort` is only used if the query is in the middle of compilation or execution 
````
if (driverState.isCompiling() || driverState.isExecuting()) {
        driverState.abort();
}
````

Reply:
yes, but the question is that the driver may be set null when the query is in the middle of compilation, and if the compiling thread just runs before lock(), the driverState will be reset to State.COMPILING from State.CLOSED OR State.DESTROYED, and keep running.